### PR TITLE
Footer and carousel unit test

### PR DESCRIPTION
**Pull request by Ne-anna**
In this PR I have created test file for footer, because it was similarly created as gallery.test.tsx file I  decided to not create separate PR, but combine it together with carousel test.
In carousel.test.tsx file I have test block which tests title and subtitle, test block which tests button with added testId, and I tried to test if button is clicked.
Issue: Even if I tested both buttons my test coverage is lower then I expected.

### CODE REVIEW

File: pages/atoms/carousel-next-button/carousel-next-button.tsx [hunk 1]
```diff
@@ -4,7 +4,11 @@ import { CarouselButton } from "../../../data";
 
 export default function CarouselNextButton(props: CarouselButton) {
   return (
-    <button className={style.carouselButton} onClick={props.loadNextSlide}>
+    <button
+      className={style.carouselButton}
+      onClick={props.loadNextSlide}
+      data-testid="buttonNext"
```
Comment (ghost) [inline pages/atoms/carousel-next-button/carousel-next-button.tsx]: rule of thumb is to use data-testid as last resort, you could add aria-label e.g. 'previous' and then use getByRole('button', {name: 'previous'} )
also afaik for buttons, the aria label / button name is determined by the text content between the button tags, this case there is image which doesnt have any text, using getByRole would solve this accessibility issue too

File: pages/atoms/carousel-previous-button/carousel-previous-button.tsx [hunk 2]
```diff
@@ -4,7 +4,11 @@ import { CarouselButton } from "../../../data";
 
 export default function CarouselPreviousButton(props: CarouselButton) {
   return (
-    <button className={style.carouselButton} onClick={props.loadPreviousSlide}>
+    <button
+      className={style.carouselButton}
+      onClick={props.loadPreviousSlide}
+      data-testid="buttonPrevious"
```
Comment (ghost) [inline pages/atoms/carousel-previous-button/carousel-previous-button.tsx]: as previously, preferably avoid data-testid, it is also rendered to the DOM (we should avoid to bloat it)

File: pages/organisms/carousel/carousel.test.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  const handleClickMock = jest.fn();
+  const { getByTestId } = render(
+    <CarouselPreviousButton loadPreviousSlide={handleClickMock} />
+  );
+  const button = screen.getByTestId("buttonPrevious");
+  fireEvent.click(button);
+  expect(handleClickMock).toHaveBeenCalledTimes(1);
+});
+
+it("Is 'previous' button clicked?", () => {
```
Comment (ghost) [inline pages/organisms/carousel/carousel.test.tsx]: duplicate test name, can mess with u if test fails

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by ilze-dom — APPROVED** [review]

File: pages/organisms/carousel/carousel.test.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+const carouselData: CarouselData = {
+  carousel: [carouselItem],
+};
+
+const carouselButtonData: CarouselButton = {
+  carouselButtonNext: "/assets/icons/icon-name.svg",
+  altTagNext: "arrow left",
+  carouselButtonPrevious: "/assets/icons/icon-name.svg",
+  altTagPrevious: "arrow right",
+  // loadPreviousSlide(): void {},
```
Comment (ilze-dom) [inline pages/organisms/carousel/carousel.test.tsx:29]: could use jest.fn() here

**Review by ilze-dom — APPROVED** [review]

**Pull request merged**